engine/cld/config/domain/domain.go

```diff
@@ -59,6 +62,13 @@ func (cfg *DomainConfig) validate() error {
 		}
 	}
 
+	// Validate JIRA config if present (it's optional)
+	if cfg.Jira != nil {
+		if err := cfg.Jira.Validate(); err != nil {
+			return errors.Join(errors.New("invalid JIRA configuration"), err)
```

Comment:
The standard way to propagate errors is to wrap the error rather than join. Joining is generally used when you want to group errors at the same level, rather than an error coming up from a call 

```suggestion
			return fmt.Errorf("invalid JIRA configuration: %w", err)
```